docs(adoptium-java datasource): rewrite (#17411)
renovatebot/renovate: lib/modules/datasource/adoptium-java/readme.md

This datasource returns releases from the [Adoptium](https://adoptium.net/) API.

> It uses `image_type=<jre|jdk>&project=jdk&release_type=ga&sort_method=DATE&sort_order=DESC&vendor=adoptium` as filter parameters.

Author: I'm not sure what `filter parameters` means here. The `image_type=<jre|jdk>&project=jdk&release_type=ga&sort_method=DATE&sort_order=DESC&vendor=adoptium` is a bit tricky to parse on first glance.

How about we add a plain language explanation like this maybe?

The datasource will check:

- JRE or JDK files
- with a JDK project
- which have the GA (general availability) status
- and sorts the results in descending order
- only checks vendor type Adoptium

Reviewer: sure, vendor filter has been removed. I Don't really know the details of the `project` filter. I use mainly the default values

https://api.adoptium.net/q/swagger-ui/#/Release%20Info/getReleaseVersions

> It only uses the first 50 pages with 50 items per page.

Author: > It only uses the first 50 pages with 50 items per page.

This is confusing.

How about something like this instead?

```suggestion
When Renovate contacts the Adoptium API, it fetches 50 pages.
Each page has 50 items.
So 2500 items are fetched from the API in total.
```

This datasource returns releases from the [Adoptium](https://adoptium.net/) API.

> It uses `image_type=<jre|jdk>&project=jdk&release_type=ga&sort_method=DATE&sort_order=DESC&vendor=adoptium` as filter parameters.

Reviewer: ```suggestion
It uses `image_type=<jre|jdk>&project=jdk&release_type=ga&sort_method=DATE&sort_order=DESC` as filter parameters.
```

This datasource returns releases from the [Adoptium](https://adoptium.net/) API.

It uses `image_type=<jre|jdk>&project=jdk&release_type=ga&sort_method=DATE&sort_order=DESC` as filter parameters.
This means that the datasource finds:

- JRE or JDK images
- with a JDK project
- which have the General Availability status

And finally, the results are sorted in descending order.

When Renovate contacts the Adoptium API, it fetches 50 pages.
Each page has 50 items.
So 2500 items are fetched from the API in total.

If you want to get releases which come with the JDK, set the `packageName` to `java-jdk` or `java`.

If you want to get releases which come with the JRE, set the `packageName` to `java-jre`.

> Currently, only LTS releases of Java come with the JRE.

Reviewer: it seems this is no longer true, i saw are for non lts too recently

Author: ```suggestion
```

I think we can drop the line then?

Reviewer: maybe we should leave something like the isn't always a jre for each version?

Reviewer 2: @viceice I think you're the only on who understands it enough to write it :)

Author: How about this then?

```suggestion
LTS releases of Java will have a version that comes with the JRE.
Non-LTS release may not always include the JRE.
```

If you want to get releases which come with the JDK, set the `packageName` to `java-jdk` or `java`.

If you want to get releases which come with the JRE, set the `packageName` to `java-jre`.
LTS releases of Java will have a version that comes with the JRE.

> Non-LTS release may not always include the JRE.

Author: ```suggestion
Non-LTS releases may not always include the JRE.
```

Whoops, it should be "releases" here. 🙈
